var/spack/repos/builtin/packages/babelstream/package.py

```diff
@@ -235,183 +464,408 @@ def cmake_args(self):
+        #               NVHPC
+        # ===================================
+        if spec.compiler.name == "nvhpc":
+            flags = pkg.compiler.opt_flags[4]  # for -O3
+            # FCFLAGS	:= -O3 -Minform=inform -Minfo=all
+            flags += " -Minform=warn "
+            TARGET = "gpu"  # target = "multicore"
+            config["TARGET"] = TARGET
+            if "cuda_arch" in self.spec.variants:
+                cuda_arch_list = self.spec.variants["cuda_arch"].value
+                # the architecture value is only number so append sm_ to the name
+                cuda_arch = "cc" + cuda_arch_list[0]
+            # config['MARCH'] = "neoverse-v1,neoverse-n1,icelake-server,znver3,cortex-a78ae"
+            GPUFLAG = " -gpu=" + cuda_arch
+            flags += "-tp=" + str(spec.target)
+            # this is to allow apples-to-apples comparison with DC in non-DC GPU impls
+            # set exactly one of these pairs!
+            # MANAGED = "-DUSE_MANAGED -gpu=managed"
+            # DEVICE=""
+            # ------------
+            DEVICE = "-DUSE_DEVICE -cuda -gpu=nomanaged"
+            MANAGED = ""
+            config["FCFLAGS"] = flags
+            config["DOCONCURRENT_FLAG"] = GPUFLAG + " -stdpar=" + TARGET + " " + DEVICE
+            config["ARRAY_FLAG"] = GPUFLAG + " -stdpar=" + TARGET + " " + MANAGED
+            config["OPENMP_FLAG"] = GPUFLAG + " -mp=" + TARGET + " " + MANAGED
+            config["OPENACC_FLAG"] = GPUFLAG + " -acc=" + TARGET + " " + MANAGED
+            config["CUDA_FLAG"] = GPUFLAG + " -cuda -acc=gpu" + " " + MANAGED
+
+        # ===================================
+        #               ONEAPI
+        # ===================================
+        if spec.compiler.name == "oneapi":
+            flags = "-std18 -Ofast -xHOST -qopt-zmm-usage=low"
+            if config["FC"] == "ifort":
+                flags += "-qopt-streaming-stores=always"
+
+            config["DOCONCURRENT_FLAG"] = "-qopenmp" + (
+                "-parallel" if config["FC"] == "ifort" else ""
+            )
+            config["ARRAY_FLAG"] = "-qopenmp" + ("-parallel" if config["FC"] == "ifort" else "")
+            config["OPENMP_FLAG"] = "-qopenmp" + (
+                "-fopenmp-targets=spir64 -DUSE_FLOAT=1" if config["FC"] == "ifx" else ""
+            )
+
+        # ===================================
+        #               FJ
+        # ===================================
+        if spec.compiler.name == "fj":
+            flags = "-X08 -Kfast -KA64FX -KSVE -KARMV8_3_A -Kzfill=100 "
+            flags += "-Kprefetch_sequential=soft "
+            flags += "-Kprefetch_line=8 -Kprefetch_line_L2=16 -Koptmsg=2 "
+            # FJ Fortran system_clock is low resolution
+            flags += "-Keval -DUSE_OMP_GET_WTIME=1 "
+
+            config["DOCONCURRENT_FLAG"] = "-Kparallel,reduction -DNOTSHARED"
+            config["ARRAY_FLAG"] = "-Kparallel,reduction"
+            config["OPENMP_FLAG"] = pkg.compiler.openmp_flag
```

Comment:
Am I missing something? It doesn't look like this `flags` variable is ever used.

Reply:
You are right, it is missing from some of the fortran models adding them too :)